### ISSUE TITLE
Add Vocabulary mapping functions

### ISSUE THREAD

**Issue by alondhe** [OHDSI/ROhdsiWebApi#25]
One request I have from my organization is to be able to map source codes in bulk to high level terms like MedDRA. The meddra branch provides a solution to this, by adding functions to go from source codes to SNOMED to MedDRA. 

Does anyone else see value in this?

**Comment by schuemie** [comment]
Sounds like this could be solved with a few generic vocabulary lookup functions? (So not specific to SNOMED to MedDRA)

**Comment by alondhe** [comment]
That's true, could allow for other lookups like to ATC. I'll look into genericizing it.

**Comment by cgreich** [comment]
> Does anyone else see value in this?

Big time. I thought we have an issue on this, have we not?

**Comment by alondhe** [comment]
@schuemie and @cgreich -- I've modified the meddra branch to be more generic. Now, you can go from a source code to a classification term, and you can specify the following parameters:

1. sourceCode: A single source code
2. sourceVocabularyId: The name of the source vocabulary id for the source code (e.g. ICD9CM, ICD10CM)
3. targetVocabularyId: The classification vocabulary id to map to
4. targetConceptClassId: (OPTIONAL) The concept class id of the classification term
5. relationshipId: The name of the concept relationship to filter by. Refer to the concept_relationship table for a list of options.
6. minDistance: The minimum distance from the standard concept for the classification term
7. maxDistance: The maximum distance from the standard concept for the classification term

Any thoughts?

**Comment by cgreich** [comment]
> 3. targetVocabularyId: The classification vocabulary id to map to

Make that optional, too.

> 6. minDistance: The minimum distance from the standard concept for the classification term
> 7. maxDistance: The maximum distance from the standard concept for the classification term

What are these distances?

**Comment by alondhe** [comment]
Those distances are to help filter the results a bit, as WebAPI provides distance values for every concept relationship. We can use these values to obtain more specific classification terms.

For instance, if we wanted only classification terms that are directly above the source code's standard concept mapping in the hierarchy, we could set min and max distance to 1. This is a use case we certainly have.

**Comment by cgreich** [comment]
Oh. That is CONCEPT_ANCESTOR relationships. Got it. Because "mapping" is usually a CONCEPT_RELATIONSHIP record.

So, which one do you want to provide?
